cumulus/pallets/xcmp-queue/src/tests.rs

```diff
@@ -341,3 +341,23 @@ fn xcmp_queue_consumes_dest_and_msg_on_ok_validate() {
 		);
 	});
 }
+
+#[test]
+fn xcmp_queue_send_xcm_works() {
+	new_test_ext().execute_with(|| {
+		let dest = (Parent, X1(Parachain(12345))).into();
+		let msg = Xcm(vec![ClearOrigin]);
+
+		// try to send without opened HRMP channel to the para_id 12345
+		assert_eq!(
+			send_xcm::<XcmpQueue>(dest, msg.clone()),
+			Err(SendError::Transport("NoChannel")),
+		);
+
+		// open HRMP channel to the para_id 12345
+		ParachainSystem::open_outbound_hrmp_channel_for_benchmarks_or_tests(12345.into());
+
+		// now send works
+		assert_ok!(send_xcm::<XcmpQueue>(dest, msg));
```

Comment:
Should be possible to also call `take_outbound_messages` and assert that the message is in the outbound buffer.

Reply:
ok, good point, we check this `take_outbound_messages` in benchmarks.
Added, do you want me to decode page/message data or just "page/message for para_id" is enough?

Reply:
Is good enough I think, thanks. I will add many more tests in https://github.com/paritytech/polkadot-sdk/pull/1246 anyway.